### DISCUSSION TITLE
Alert enhancements

### DISCUSSION THREAD

**Discussion by basham** [indiana-university/rivet-source#546] [Feature and enhancement requests]
The following are some suggestions for improving the [Alert components](https://rivet.iu.edu/components/overlays/alerts/).

**Comment by basham** [comment]
Page alerts need a title, but not all page alerts need a message. However, the styles for it assumes there will always be a message. `rvt-alert__title` includes `margin-bottom: 0.25rem;`. However, this should be replaced with `.rvt-alert__message { margin-top: 0.25rem; }`. This means that the spacing is only included if a message is used.

**Comment by basham** [comment]
The use of background colors and colored icons seems like sufficient ways to visually hint at the type of the alert. I'm not convinced we need colored text for alerts. Using the standard body black color will create more contrast; reduce the number of colors needed in the Rivet system; and look more normal if the content includes the standard blue links. And we can also consider adding a full border around the alerts, rather than just the left side. That will also maintain good contrast, with whatever the surrounding context is. I explore these options in this [alert styles CodePen](https://codepen.io/basham/pen/YzVPOxL).

**Comment by basham** [comment]
The line height for `.rvt-alert__title` calculates to 21.6px. It would be better if the line height was 24px, so that it aligns perfectly with the size of the dismiss button (24&times;24px).

**Comment by levimcg** [comment]
@basham Thanks for creating this discussion, we really appreciate the thought and time you've put into this! We're getting ready to run into a slow time of year were lots of folks will be away on PTO, but I want to definitely get some discussion started around some of the items in this thread (as well as the others you opened: #545 #544 #543 ).

We met last week as a team to review and there are definitely some things in this thread that we would like to move forward with. It's really more a matter of planning and prioritizing the work. I'm going to do my best to start commenting/starting a thread for each of these items and bringing back some of what the team talk about when we reviewed these items.

**Comment by basham** [comment]
[Alert lists](https://rivet.iu.edu/components/overlays/alerts/#alert-lists) needs to be reviewed. The documentation shows a list of page-level error alerts. However, if there are multiple errors on a page (such as a multiple validation errors in a form after submitting), it seems like it would be best served as a single page-level error alert. The alert message would include a list of errors. In the case of form validation, the list of errors would be anchor links to the form controls with the described error.

A list of page-level alerts as it is presented may be appropriate if they are all of different alert types. For example, maybe you submit a form that bulk adds some records. You could accept 20 of the records that were entered correctly and show a success alert. And you could reject 5 records that were entered incorrectly and show an error alert. So, in this case, you'd show one success alert and one error alert.
